project/Build.scala

```diff
@@ -433,6 +433,49 @@ object Build {
 
   private lazy val currentYear: String = java.util.Calendar.getInstance().get(java.util.Calendar.YEAR).toString
 
+  private val shellBanner: String =
+    """%n      ________ ___   / /  ___
+       |%n    / __/ __// _ | / /  / _ |
+       |%n  __\\ \\/ /__/ __ |/ /__/ __ |
+       |%n /____/\\___/_/ |_/____/_/ | |
+       |%n                          |/  %s""".stripMargin.replace("\n", "")
+
+  // Common generator for properties files
+  lazy val generatePropertiesFile = (contents: String, fileName: String) => Def.task {
+    val file = (Compile / resourceManaged).value / fileName
+    if (!(file.exists && IO.read(file) == contents)) {
+      IO.write(file, contents)
+    }
+    Seq(file)
+  }
+
+  // Generate compiler.properties consumed by sbt
+  lazy val generateCompilerProperties: Def.Initialize[Task[Seq[File]]] = {
+    import java.util._
+    import java.text._
+    val fileName = "compiler.properties"
+    val dateFormat = new SimpleDateFormat("yyyyMMdd-HHmmss")
+    dateFormat.setTimeZone(TimeZone.getTimeZone("GMT"))
+    val contents =                //2.11.11.v20170413-090219-8a413ba7cc
+      s"""version.number=${version.value}
+          |maven.version.number=${version.value}
+          |git.hash=${VersionUtil.gitHash}
+          |copyright.string=Copyright 2002-$currentYear, LAMP/EPFL
+        """.stripMargin
+    generatePropertiesFile(contents, fileName)
+  }
+
+  // Generate library3.properties consumed by scala.util.Properties
+  lazy val generateLibrary3Properties: Def.Initialize[Task[Seq[File]]] = {
+    val fileName = "library3.properties"
```

Comment:
I think we should generate just the `library.properties` file, not `library3.properties` 
There would never be any clash - we're emitting the `scala-library` JAR, there won't be any `scala-library:2.x` jar on the classpath, thanks to eviction the `scala-library:3.8` would be used.  

`library3.properties` would make sense in the `scala3-library_3` artifact, but now it's just an empty jar with dependency on `scala-library`